crates/builder/src/tasks/block.rs

```diff
@@ -105,6 +156,13 @@ impl BlockBuilder {
                         _ = &mut sleep => {
                             if !in_progress.is_empty() {
                                 tracing::debug!(txns = in_progress.len(), "sending block to submit task");
+                                // inget from tx pool
+                                if let Ok(tx_list) = self.tx_poller.poll_transactions().await {
```

Comment:
currently this will only run if at least 1 user submitted a tx. instead, we want it to be an upstream task use the same channel to send txns into this task. this means defining a spawn loop for the poller, and doing task setup in the same part of `main.rs` as the other task loops